### PR TITLE
deepin.dde-launcher: init at 4.6.13

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

Add [dde-launcher](https://github.com/linuxdeepin/dde-launcher), the launcher of Deepin Desktop Environment.

About packaging deepin for NixOS: #59023

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by romildo** [comment]
Tested on a virtual machine.

Screenshot from 2019-06-22 18-29-41

**Comment by romildo** [comment]
@GrahamcOfBorg build deepin.dde-launcher

File: pkgs/desktops/deepin/dde-launcher/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{ stdenv, fetchFromGitHub, pkgconfig, cmake, dde-qt-dbus-factory,
+  deepin, deepin-desktop-schemas, deepin-wallpapers, dtkcore,
+  dtkwidget, gsettings-qt, qtsvg, qttools, qtx11extras, wrapGAppsHook
+  }:
```
Comment (worldofpeace) [inline pkgs/desktops/deepin/dde-launcher/default.nix]:
```suggestion
}:
```

**Review by worldofpeace — CHANGES_REQUESTED** [review]
There's a few executables used with [`QProcess`](https://github.com/linuxdeepin/dde-launcher/search?q=QProcess&unscoped_q=QProcess) that we should hardcode.

**Comment by romildo** [comment]
> There's a few executables used with [`QProcess`](https://github.com/linuxdeepin/dde-launcher/search?q=QProcess&unscoped_q=QProcess) that we should hardcode.

Done.

File: pkgs/desktops/deepin/dde-launcher/default.nix (hunk 1 above)
Comment (romildo) [reply pkgs/desktops/deepin/dde-launcher/default.nix]: Fixed.

File: pkgs/desktops/deepin/dde-launcher/default.nix [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    substituteInPlace CMakeLists.txt --replace "/usr/share" "$out/share"
+
+    substituteInPlace src/dbusservices/com.deepin.dde.Launcher.service --replace "/usr" "$out"
+
+    substituteInPlace src/historywidget.cpp --replace "xdg-open" "${xdg_utils}/bin/xdg-open"
+    substituteInPlace src/widgets/miniframebottombar.cpp --replace "dbus-send" "${dbus}/bin/dbus-send"
+    substituteInPlace src/widgets/miniframebottombar.cpp --replace "dde-shutdown" "${dde-session-ui}/bin/dde-shutdown"
+    substituteInPlace src/widgets/miniframenavigation.cpp --replace "dbus-send" "${dbus}/bin/dbus-send"
+    substituteInPlace src/widgets/miniframerightbar.cpp --replace "which" "${which}/bin/which"
+    substituteInPlace src/widgets/searchwidget.cpp --replace "dbus-send" "${dbus}/bin/dbus-send"
```
Comment (worldofpeace) [inline pkgs/desktops/deepin/dde-launcher/default.nix]: https://github.com/linuxdeepin/dde-launcher/blob/523b1feb0638f7fe23ee6519df8a873b368d8c55/src/widgets/searchwidget.cpp#L81 should return true so this portion of code wouldn't be ran right?

Comment (romildo) [reply pkgs/desktops/deepin/dde-launcher/default.nix]: `DTK_VERSION >= DTK_VERSION_CHECK(2, 0, 8, 0))`

As dtkcore/dtkwidget current version is 2.0.14, this test is true,  `dbus-send` will not be explicitly run.
Therefore `dbus-send` path does not need to be hard coded, right?
I  am going to remove them from the nix expression.

Comment (worldofpeace) [reply pkgs/desktops/deepin/dde-launcher/default.nix]: Yep that is what I meant :+1:

**Review by worldofpeace — APPROVED** [review]

**Pull request merged**